### PR TITLE
Added optional public network interface

### PR DESCRIPTION
**Pull request by dcwangmit01**
* Based on teh network setting in config.yaml.example, a second
  network interface will be created and set as the default route

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -101,18 +139,10 @@ Vagrant.configure("2") do |config|
...
-    kdk_dev_folder = YAML.load_file(config_file)['kdk_dev_folder']
-  else
-    kdk_dev_folder = "Dev"
-  end
 
-  if File.directory?(File.expand_path("~/" + kdk_dev_folder))
-    config.vm.synced_folder "~/" + kdk_dev_folder, "/home/vagrant/Dev"
+  host_dev_folder = if custom != nil && custom.key?('host_dev_folder') then custom['host_dev_folder'] else "~/Dev" end
+  if File.directory?(File.expand_path(host_dev_folder))
+    config.vm.synced_folder host_dev_folder, "/home/vagrant/Dev"
```
Comment (rtluckie) [inline Vagrantfile:145]: NIT: is it possible to use the same name as the custom `host_dev_folder` rather than default to `Dev`?

**Review by rtluckie — APPROVED** [review]

File: Vagrantfile (hunk 1 above)
Comment (dcwangmit01) [reply Vagrantfile:145]: @rtluckie I'm going to merge this PR, and then follow-on with a PR that makes the host_dev_folder a list of from_host_dirs to_guest_dirs mounts.  I'll also change the config.yaml.example to be config.yaml.defaults, which can still be overridden via a config.yaml.  Then I can remove some of the logic to check that config.yaml exists ,and remove the default hard coding in this file.  Cool?

Comment (rtluckie) [reply Vagrantfile:145]: sgtm

Comment (dcwangmit01) [reply Vagrantfile:145]: Thanks @rtluckie .  Merging then.  Will follow-on a PR

Comment (dcwangmit01) [reply Vagrantfile:145]: @rtluckie  See this follow-on PR

https://github.com/cisco-sso/k8s-devkit/pull/13

**Pull request merged**